os/binfmt/binfmt_unloadmodule.c

```diff
@@ -189,15 +190,15 @@ int unload_module(FAR struct binary_s *binp)
 			}
 
 #endif
-			for (int section_idx = 0; section_idx < BIN_MAX; section_idx++) {
-				binp->sections[section_idx] = NULL;
+			for (section_idx = 0; section_idx < BIN_MAX; section_idx++) {
+				binp->sections[section_idx] = 0;
```

Comment:
Regardless of this PR, do we need to run the loop again in CONFIG_BINFMT_SECTION_UNIFIED_MEMORY disabled? 
I think we can merge two loops.
@kishore-sn